### PR TITLE
Bug Fix: Either.tryCatch now refines the error

### PR DESCRIPTION
**Pull request by gcanti**

### CODE REVIEW

**Comment by gcanti** [comment]
@raveclassic thanks!

File: src/Either.ts [hunk 1]
```diff
@@ -281,7 +281,11 @@ export const tryCatch = <A>(f: Lazy<A>): Either<Error, A> => {
   try {
     return right(f())
   } catch (e) {
-    return left(e)
+    if (e instanceof Error) {
+      return left(e)
+    } else {
+      return left(new Error(String(e)))
```
Comment (OliverJAsh) [inline src/Either.ts]: What happens if the error thrown is an object, which is so often the case is badly written JS libs? `String({ message: 'foo' }) // => [object Object]` 😒

Comment (raveclassic) [reply src/Either.ts]: This exactly what I was pointing to in the issue #323

Comment (raveclassic) [reply src/Either.ts]: Anyway, this bug is about incorrect return type Error for things that actually aren't Errors in tryCatch. We still need some proper solution to proccess incorrect thrown entities.

**Pull request merged**